patches/minecraft/net/minecraft/network/play/server/SCommandListPacket.java.patch

```diff
@@ -0,0 +1,10 @@
+--- a/net/minecraft/network/play/server/SCommandListPacket.java
++++ b/net/minecraft/network/play/server/SCommandListPacket.java
+@@ -221,6 +_,7 @@
+                   this.field_197726_a.executes((p_197724_0_) -> {
+                      return 0;
+                   });
++                  this.field_197726_a.executes(null);
```

Comment:
Why?

Reply:
I'm not the author of the PR, but after reading the original PR and the justification given by the original PR's author (https://github.com/MinecraftForge/MinecraftForge/pull/6952#discussion_r458315650), I believe I understand the purpose of this.

**My suggestion**: As said in the linked review comment from the original PR, please add a comment briefly explaining why Forge sets the `executes` lambda of commands to be `null`.

***semi TL;DR: This allows the client hook to know that there is no client-side command, so it sends the command over to the server for final processing.***

### Background

This PR allows clients to register their custom commands to the client-side `CommandDispatcher<ISuggestionsProvider>` (as opposed to the server-side `CommandDispatcher<CommandSource>`). But, the commands are usually sent over to the server for final processing. 

### PR Logic

To allow the client-side custom commands to run, the PR manually executes the client-side dispatcher with the command. If there is no valid command found, then it delegates to the server for final execution. This happens when the parsed command tree does not end in a valid `executes` lambda. When no valid command is found, a `CommandSyntaxException` with the [`dispatcherUnknownCommand()`](https://github.com/Mojang/brigadier/blob/cf754c4ef654160dca946889c11941634c5db3d5/src/main/java/com/mojang/brigadier/exceptions/BuiltInExceptionProvider.java#L59) type is thrown. 

The hook captures command syntax exceptions, and if it captures one [with this specific type](https://github.com/coehlrich/MinecraftForge/blob/4b0d0484a6c6dbb5e293f2413b21909466d63ec8/src/main/java/net/minecraftforge/client/ForgeHooksClient.java#L854-L857), it assumes that there is no client-side command (otherwise execution would have succeeded, or an exception without that specific type is thrown), and thus throws it over to the server for processing.

(See [`ForgeHooksClient#sendMessage`](https://github.com/coehlrich/MinecraftForge/blob/4b0d0484a6c6dbb5e293f2413b21909466d63ec8/src/main/java/net/minecraftforge/client/ForgeHooksClient.java#L841) for the logic of the hook.)

### Modifying Vanilla's Behavior

In the original vanilla logic, when the client receives the server-side command information to construct a replica of the command tree (stored in the `CommandDispatcher<ISuggestionsProvider>`), it adds in empty lambdas for the command trees. This means that the explained logic would never work: since the command is valid due to the existence of the `executes` lambda, it gets executed _on the client_ (as the logic assumes if a command executes, it must be a custom client-side command) and it is never sent to the server for actual execution.

To allow the logic of the PR to work, this patch sets the `executes` parts of the replica command tree to be `null` instead of an empty (executing) lambda. This causes the command dispatcher to throw a command syntax exception with the aforementioned type, which allows the hook's logic to intercept it and send the command over to the server.

### Alternative

Possibly, this PR could create its own command syntax exception type, and instead of `null`-ing the lambdas, throws the exception with that custom type, and the hook would recognize that type specifically as the marker to send the command over to the server for real processing.

This has the added benefit of only sending valid server-listed commands (as defined in the replica command tree) over to the server for processing, while swallowing and presenting to the user if there was a genuine error with the client-side custom command (such as accidentally forgetting to add an `executes`).